CoderSdk/Coder/WorkspaceAgents.cs

```diff
@@ -0,0 +1,38 @@
+namespace Coder.Desktop.CoderSdk.Coder;
+
+public partial interface ICoderApiClient
+{
+    public Task<WorkspaceAgent> GetWorkspaceAgent(string id, CancellationToken ct = default);
+}
+
+public class WorkspaceAgent
+{
+    public const string DisplayAppVscode = "vscode";
+    public const string DisplayAppVscodeInsiders = "vscode_insiders";
+
+    public string ExpandedDirectory { get; set; } = "";
+
+    public WorkspaceApp[] Apps { get; set; } = [];
+
+    // This isn't an enum to avoid future display apps breaking the desktop
+    // app.
+    public string[] DisplayApps { get; set; } = [];
+}
+
+public class WorkspaceApp
+{
+    public string Id { get; set; } = string.Empty;
+    public string Url { get; set; } = string.Empty;
+    public bool External { get; set; } = false;
+    public string DisplayName { get; set; } = string.Empty;
+    public string Command { get; set; } = string.Empty;
+    public string Icon { get; set; } = string.Empty;
+}
+
+public partial class CoderApiClient
+{
+    public Task<WorkspaceAgent> GetWorkspaceAgent(string id, CancellationToken ct = default)
```

Comment:
This would be safer if we accepted a UUID, rather than a string, which could totally change the endpoint we hit.

Reply:
I considered this but didn't want to add a dependency between `CoderSdk` and `Vpn.Proto`, and adding a new project for just a UUID type seems overkill. The Go codersdk package accepts string IDs for some methods as well, so I think it's fine.

Reply:
I had (wrongly) assumed/remembered that we parsed the `Agent` protobufs into a model with `Guid` id types. But, we just leave it as a byte string. You're right, not worth introducing the dep.

Reply:
Actually, is there some reason we need to roll our own `Uuid` class? I believe https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-9.0 is the same thing, but MSFT uses a different name than other systems.

Reply:
Could we move our `Uuid` implementation to CoderSdk? There will likely be plenty of times we want to interact with UUIDs outside of the protobuf protocols.